src/add-archival-events/src/db.ts

```diff
@@ -0,0 +1,163 @@
+import groupBy from "lodash.groupby"
+import type { QueryResult } from "pg"
+import { Client } from "pg"
+import { logger } from "shared"
+import type { PromiseResult } from "shared-types"
+
+export interface Dictionary<T> {
+  [Key: number]: T
+}
+
+export type BichardRecord = {
+  messageId: string
+  recordId: number
+  archivedAt: Date
+  archivedBy: string
+  groupId: number
+}
+
+export type DatabaseRow = {
+  message_id: string
+  error_id: number
+  archived_at: string
+  archived_by: string
+  archive_log_id: number
+}
+
+// Produces the string "$1, $2, $3..." for the given range
+const wherePlaceholderForRange = (range: number) => [...Array(range).keys()].map((x) => `$${x + 1}`).join(", ")
+
+export default class DatabaseClient {
+  private postgres: Client
+
+  constructor(
+    private host: string,
+    private user: string,
+    private password: string,
+    private useSsl: boolean,
+    private database: string,
+    private schema: string,
+    private archiveGroupLimit: number
+  ) {
+    this.postgres = new Client({
+      host: this.host,
+      user: this.user,
+      password: this.password,
+      ssl: this.useSsl
+        ? {
+            rejectUnauthorized: true
```

Comment:
@emadkaramad @daviesjamie this is what we were chatting about recently and didn't we come to the conclusion that it made more sense (introduce less work) to have this set to false?

We don't need to verify the authenticity of a cert within our private vpc, yes there is an opportunity for an attack where the attacker spoofs our dbs cert and convinces our services it's the real db but at that point they have compromised the vpc and creating a fake cert would be the least of our worries.

Setting this to true could cause any self signed certs to fail. Because we control the network it's fine to have this set to false.

Reply:
ah right, I didn't realise we used self-signed certs for the database in prod but that would cause a problem with this as-is, and you're right that it's a very marginal security benefit when the VPC is so locked down 👍